src/cloudformation_cli_python_lib/resource.py

```diff
@@ -60,6 +70,44 @@ def _add_handler(f: HandlerSignature) -> HandlerSignature:
 
         return _add_handler
 
+    @staticmethod
+    def schedule_reinvocation(
+        handler_request: HandlerRequest,
+        handler_response: ProgressEvent,
+        context: LambdaContext,
+        credentials: Credentials,
+    ) -> bool:
+        if handler_response.status != OperationStatus.IN_PROGRESS:
+            return False
+        # modify requestContext dict in-place, so that invoke count is bumped on local
+        # reinvoke too
+        reinvoke_context = handler_request.requestContext
+        reinvoke_context["invocation"] = reinvoke_context.get("invocation", 0) + 1
+        callback_delay_s = handler_response.callbackDelaySeconds
+        remaining_ms = context.get_remaining_time_in_millis()
+        needed_ms_remaining = callback_delay_s * 1200 + INVOCATION_TIMEOUT_MS
```

Comment:
this needs the comments from the java plugin; otherwise 1200 makes 0% sense

Reply:
Future travellers; 1000 (ms in s) * 120%